File: 2284-smallest-value-of-the-rearranged-number/smallest-value-of-the-rearranged-number.cpp
class Solution {
public:
    long long smallestNumber(long long num) {
        string s=to_string(num);
        if(s[0]=='-'){
             sort(s.begin()+1, s.end(), greater<char>());
             return stoll(s);
        }
        sort(s.begin(),s.end());
        if(s[0]!='0'){
            return stoll(s);
        }
        
        for(int i=1;i<s.length();i++){
            if(s[i]!='0'){
                swap(s[0],s[i]);
                return stoll(s);
            }
        }
        return 0;
    }
};

File: 2284-smallest-value-of-the-rearranged-number/smallest-value-of-the-rearranged-number.py
class Solution:
    def smallestNumber(self, num: int) -> int:
        if num == 0:
            return 0
        
        digits = list(str(abs(num)))
        
        if num < 0:
            digits.sort(reverse=True)
            return -int(''.join(digits))
        else:
            digits.sort()
            if digits[0] != '0':
                return int(''.join(digits))
            else:
                for i in range(1, len(digits)):
                    if digits[i] != '0':
                        digits[0], digits[i] = digits[i], digits[0]
                        break
                return int(''.join(digits))



       
        